File: src/0268-Missing-Number/missing_number.py
from typing import List

class Solution:
    def missingNumber(self, nums: List[int]) -> int:
        nums_len: int = len(nums)
        total_sum: int = 0
        missing: int = 0
        
        for i in range(nums_len):
            total_sum += i
            missing += nums[i]

        return total_sum - missing + nums_len


if __name__ == "__main__":
    nums: List[int] = [9,6,4,2,3,5,7,0,1]
        
    solution = Solution()
    
    print(solution.missingNumber(nums))

File: src/0268-Missing-Number/missing_number.cpp
#include <iostream>
#include <vector>

class Solution {
public:
    int missingNumber(std::vector<int>& nums) {
        int nums_len = nums.size();
        int total_sum = 0;
        int missing = 0;

        for (int i = 0; i < nums_len; i++) {
            total_sum += i;
            missing += nums[i];
        }

        return total_sum - missing + nums_len;
    }
};

int main() {
    std::vector<int> nums = {9,6,4,2,3,5,7,0,1};
    
    Solution solution;

    std::cout << solution.missingNumber(nums);
    
    return 0;
}